docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+  ``get_dataframe`` for DataFrame).
+
+  2. process that data in some way, optionally by calling an external workflow function.
+  3. rename or update attributes from HydroMT variable conventions (name, unit) to the specific model conventions.
+  4. adds the data to the corresponding HydroMT model components.
+
+A simplified example of what a setup function would look like for example to add a landuse grid from a raster input
+data using an external workflow from hydromt core is:
+
+.. code-block:: python
+
+  def setup_landuse(
+        self,
+        landuse_fn: Union[str, Path, xr.DataArray],
+    ):
+        """Add landuse data variable to grid.
+
+        Adds model layers:
+
+        * **landuse_class** grid: data from landuse_fn
+
+        Parameters
+        ----------
+        landuse_fn: str, Path, xr.DataArray
+            Data catalog key, path to raster file or raster xarray data object.
+            If a path to a raster file is provided it will be added
+            to the data_catalog with its name based on the file basename without
+            extension.
+        """
+        self.logger.info(f"Preparing landuse data from raster source {landuse_fn}")
+        # 1. Read landuse raster data
+        da_landuse = self.data_catalog.get_rasterdataset(
+            landuse_fn,
+            geom=self.region,
+            buffer=2,
+            variables=["landuse"],
+        )
+        # 2. Do some transformation or processing
+        ds_out = hydromt.workflows.grid.grid_from_rasterdataset(
+            grid_like=self.grid,
+            ds=da_landuse,
+            fill_method="nearest",
+            reproject_method="mode",
+        )
+        # 3. Rename or transform from HydroMT to model conventions
+        rmdict = {"landuse": "landuse_class"}
+        # Or using a properly initialised _GRIDS
+        # rmdict = {k: v for k, v in self._GRIDS.items() if k in ds_out.data_vars}
+        ds_out = ds_out.rename(rmdict)
+
+        # 4. Add to grid
+        self.set_grid(ds_out)
+
+.. NOTE::
+
+  **Input data type of the setup method**: Typically a setup function tries to go from one type of dataset
+  (landuse raster) to a HydroMT model component (landuse map in ``maps``). So it's good to make clear for your user in
+  the setup function docstrings which type of input data this function can work with. You could decide to support
+  several data types in one setup function but be aware that the GIS processing functions like resampling, reprojection can
+  be quite different for a raster or a vector for example. So you could decide to create two setup functions that
+  prepare the same data but from different type of input data (eg *setup_landuse_from_raster* and *setup_landuse_from_vector*).
+
+
+Base or region setup method(s)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When building a model from scratch, the first thing you should do is to define the region in the world your model is located in,
```

Comment:
I think the `setup_grid`, `setup_mesh`, etc methods to define the geospatial representation of the model core computational unit is the first step. This typically (but not always) takes the region as an argument.

Reply:
Perhaps this should also be moved to before the setup methods section as that's the more logical order I think.